pkg/azuredisk/controllerserver_v2.go

```diff
@@ -246,8 +247,26 @@ func (d *DriverV2) ControllerPublishVolume(ctx context.Context, req *csi.Control
 
 	volumeCapability := generateAzVolumeCapability(volCap)
 
+	attachMc := metrics.NewMetricContext(d.cloudProvisioner.GetMetricPrefix(), "attach_volume_latency", d.cloudProvisioner.GetCloud().ResourceGroup, d.cloudProvisioner.GetCloud().SubscriptionID, d.Name)
+	isAttachSuccessful := false
+
 	response, err := d.crdProvisioner.PublishVolume(ctx, diskURI, nodeID, &volumeCapability, req.GetReadonly(), req.GetSecrets(), req.GetVolumeContext())
 
+	// Go Routine to calculate the attach_volume_latency
+	if err == nil {
```

Comment:
The dispatch of the goroutine can be moved after the error check and return statement below. This would avoid the redundant check of `err` and the possibility that someone inserts code between the goroutine dispatch and real error return check.

Reply:
Done.